api/apps/geoprocessing/src/marxan-sandboxed-runner/adapters/solutions-output/result-parser.service.ts

```diff
@@ -81,46 +75,12 @@ export class ResultParserService {
             if (row === '') {
               return;
             }
-            const [
-              runId,
-              score,
-              cost,
-              planningUnits,
-              connectivity,
-              connectivityTotal,
-              connectivityIn,
-              connectivityEdge,
-              connectivityOut,
-              connectivityInFraction,
-              penalty,
-              shortfall,
-              missingValues,
-              mpm,
-            ] = row.split(',');
-            const entry = plainToClass<ResultRow, ResultRow>(ResultRow, {
-              runId: +runId,
-              score: +score,
-              cost: +cost,
-              planningUnits: +planningUnits,
-              connectivity: +connectivity,
-              connectivityTotal: +connectivityTotal,
-              connectivityIn: +connectivityIn,
-              connectivityEdge: +connectivityEdge,
-              connectivityOut: +connectivityOut,
-              connectivityInFraction: +connectivityInFraction,
-              penalty: +penalty,
-              shortfall: +shortfall,
-              missingValues: +missingValues,
-              mpm: +mpm,
-              best: false,
-              // distinctFive should always be set to `false` here: the five most
-              // different solutions are tagged as such via
-              // `MostDifferentService`.
-              distinctFive: false,
-              puValues: planningUnitSelection['1'].values.map((b) =>
-                b ? 1 : 0,
-              ),
-            });
+
+            const entry = ResultParserService.calculatePuValues(
+              ResultParserService.parseRowToResultRow(row),
+              planningUnitSelection,
```

Comment:
I may be not noticing something obvious - sorry 😬 - is there any reason why we cannot "just" assign that value within `parseRowToResultRow`? If possible, would be a bit more straightforward to understand.

Reply:
I was aiming at maintaining the existing behavior, where we only set `puValues` in the second call of this function (line 120 of the old file), whereas in the initial call (line 64 of the old file), `puValues` was initialized as empty - but I agree this way doing this is more confusing :upside_down_face:

I'll apply your suggestion and I'll set `puValues` straight away on `parseRowToResultRow` and check the tests - if nothing breaks, we should be ok doing this :) (unless ofc you think we should keep the previous behavior!)